watertap/core/zero_order_costing.py

```diff
@@ -0,0 +1,510 @@
+            self.laboratory_cost, self.laboratory_cost_constraint)
+        calculate_variable_from_constraint(
+            self.insurance_and_taxes_cost,
+            self.insurance_and_taxes_cost_constraint)
+
+        calculate_variable_from_constraint(
+            self.total_fixed_operating_cost,
+            self.total_fixed_operating_cost_constraint)
+
+    def add_LCOW(self, flow_rate):
+        """
+        Add Levelized Cost of Water (LCOW) to costing block.
+
+        Args:
+            flow_rate - flow rate of water (volumetric) to be used in
+                        calculating LCOW
+        """
+        self.LCOW = pyo.Expression(
+            expr=(self.total_capital_cost*self.capital_recovery_factor +
+                  self.total_operating_cost) /
+                 (pyo.units.convert(
+                     flow_rate,
+                     to_units=pyo.units.m**3/self.base_period) *
+                  self.utilization_factor),
+            doc='Levelized Cost of Water')
+
+    def add_electricity_intensity(self, flow_rate):
+        """
+        Add calculation of overall electricity intensity to costing block.
+
+        Args:
+            flow_rate - flow rate of water (volumetric) to be used in
+                        calculating electricity intensity
+        """
+        self.electricity_intensity = pyo.Expression(
+            expr=pyo.units.convert(
+                self.aggregate_flow_electricity / flow_rate,
+                to_units=pyo.units.kWh/pyo.units.m**3),
+            doc='Overall electricity intensity')
+
+    # -------------------------------------------------------------------------
+    # Unit operation costing methods
+    def cost_power_law_flow(blk):
+        """
+        General method for costing equipment based on power law form. This is
+        the most common costing form for zero-order models.
+
+        CapCost = A*(F/Fref)**B
+
+        This method also registers electricity demand as a costed flow (if
+        present in the unit operation model).
+        """
+        t0 = blk.flowsheet().time.first()
+        ZeroOrderCostingData._general_power_law_form(blk, time=t0)
+
+        # Register flows
+        blk.config.flowsheet_costing_block.cost_flow(
+            blk.unit_model.electricity[t0], "electricity")
+
+    def cost_chemical_addition(blk):
+        """
+        General method for costing chemical addition processes. Capital cost is
+        based on the mass flow rate of chemcial added.
+
+        This method also registers the chemical flow and electricity demand as
+        costed flows.
+        """
+        chem_name = blk.unit_model.config.process_subtype
+
+        t0 = blk.flowsheet().time.first()
+        chem_flow_mass = (blk.unit_model.chemical_dosage[t0] *
+                          blk.unit_model.properties[t0].flow_vol /
+                          blk.unit_model.ratio_in_solution)
+        sizing_term = chem_flow_mass / (pyo.units.lb/pyo.units.day)
```

Comment:
Sort of hard for me to follow but does ``sizing_term`` go to gal/day at some point? Most (if not all) of the costing relationships for chemical additions are based off solution flow in gal/day.

Reply:
Not sure why I did this either - I think I got confused by the units for costing the flow versus the capital cost.